### ISSUE TITLE
Is the implicit get after a put necessary?

### ISSUE THREAD

**Issue by pn-santos** [telia-oss/github-pr-resource#32]
Is the implicit `get` that concourse does after a `put` is necessary for this resource to work? or could it be skipped?

I've had a look at the source code and there seems to be no way to use something similar to the `get_params: {skip_download: true}` of the docker image resource to skip the implicit `get` step (please correct me if I'm wrong).

In my case, I'm running 4 "checks" concurrently and at first glance, the get that is always executed after I update the checks status (with `put`) seems unecessary... Would it make sense to check for a "skip" in the `in` step?

**Comment by itsdalmo** [comment]
Thanks for submitting and issue! I totally agree with you that we should include a `skip_download` parameter and perhaps default it to `true` (since we don't push code with this resource anyway, just comments and/or statuses). 

I'll want to verify whether Concourse actually does a `get` after each `put` first, since it might be smart enough to reuse the container from the initial `get` when the subsequent `put`'s don't emit new versions. Will report back here!

**Comment by pn-santos** [comment]
If it helps this is what I see for implicit gets:
```
Initialized empty Git repository in /tmp/build/get/.git/
Switched to a new branch '8697f5223bb5fa865736aece316748192a7c22fe'
Updating 8697f52..95a1afa
Fast-forward
```

**Comment by itsdalmo** [comment]
I think we'll run into a problem if we try to skip cloning the repo on the implicit `get` after a `put`; the `get` will still run and if we don't fetch the pull request, the `pull-request` volume will be empty when it's time to run our tests:

```yml
jobs:
- name: test
  plan:
  - get: pull-request
    trigger: true
    version: every
  - put: pull-request
    params:
      path: pull-request 
      status: pending 
    get_params: { skip_download: true } <--- pull-request volume will be empty!
  - task: test
    input_mapping: { source: pull-request } <--- Oh no!
```

I've asked in the Concourse discord to validate whether or not this is how it works 😓

**Comment by pn-santos** [comment]
ups sorry closed by accident eheh, alright I'll wait for the feedback from the concourse team

**Comment by vito** [comment]
@itsdalmo Yeah, if you implement `skip_download` it'll break that example. I've had to "alias" the step to prevent that, something like:

```yaml
put: update-status
resource: pull-request
params: {...}
get_params: {skip_download: true}
```

This will still have the implicit `get` but its (empty) bits will be named `update-status` rather than clobbering the original `pull-request` bits.

Which is super janky, but well, all PR workflows will be until we finish spaces support (and perhaps a notifications API for communicating commit status would be nice). :)

**Comment by itsdalmo** [comment]
Thanks @vito! Looking forward to resources v2, but what you are suggesting seems clean enough for now 😄 What do you think @pn-santos ?

**Comment by pn-santos** [comment]
So if I understand correctly, I would need to "alias" the `put`s for which I want to skip the implicit get _and_ I would need to add the `get_params`. 

If that's so seems like an acceptable compromise.

I was actually already thinking of aliasing them anyway. I'm using this resource to set 5 separate checks on github and having parallel `put`s (3 for each check, 1 pending, 1 for failure and another for success) makes for a very confusing build log since it will cause the pull request resource name to be repeated all over the place. Aliasing makes it easier to distinguish them.

**Comment by itsdalmo** [comment]
@pn-santos - I've merged the PR to add `skip_download` as a parameter to `get` and it should be available under the `dev` tag any minute: https://hub.docker.com/r/teliaoss/github-pr-resource/tags/

Would you mind taking it for a spin and seeing if it works? I'm travelling today so unfortunately cannot test it in the wild myself 😄

**Comment by pn-santos** [comment]
Sure I'll report back the results, thx for the quick turn around :+1:

**Comment by pn-santos** [comment]
Everything seems to work ok, only had 1 minor issue, when using:
```
get_params: { skip_download: true }
```
I got this:
```
2018/09/19 20:32:23 failed to unmarshal request: json: cannot unmarshal bool into Go struct field GetParameters.skip_download of type string
```

I guess `SkipDownload` [type](https://github.com/telia-oss/github-pr-resource/pull/33/files#diff-dfd8e4ba70b1bf49b7b7d7b33ca5801cR89) should be `bool` instead of string?

When I switched to:
```
get_params: { skip_download: "true" }
```
then everything worked :100:

**Comment by itsdalmo** [comment]
Thanks for testing this @pn-santos - #37 made `skip_download` into a boolean and has been released under the `v0.6.0` tag.
